lib/dns.js

```diff
@@ -311,26 +318,31 @@ function setServers(servers) {
   const orig = cares.getServers();
   const newSet = [];
   const IPv6RE = /\[(.*)\]/;
-  const addrSplitRE = /:\d+$/;
+  const addrSplitRE = /(^.+?)(?::(\d+))?$/;
```

Comment:
I think keeping RegEx would be fine because URL does some extra operation in semantics, though URL's performance may better than RegEx.